sprint.rb

```diff
@@ -0,0 +1,56 @@
+def factorial_r(n)
+  return 1 if n == 0 || n == 1
+
+  n * factorial_r(n - 1)
+end
+
+def factorial_i(n)
+  return 1 if n == 0 || n == 1
+
+  result = 1
+  1.upto(n) { |i| result *= i }
+
+  result
+end
+
+def sumdig_r(n, result = 0)
+  if n == 0
+    return result
```

Comment:
`return n if n < 10` 
If the input is a single digit number then the sum will be the number itself.